flycheck.el

```diff
@@ -2173,10 +2200,45 @@ into the verification results."
         (princ (make-string (- message-column (current-column)) ?\ ))
         (let ((message (flycheck-verification-result-message result))
               (face (flycheck-verification-result-face result)))
-          (insert (propertize message 'face face)))
+          ;; If face is nil, using propertize erases the face already contained
+          ;; by the message.  We don't want that, since this would remove the
+          ;; button face from the checker chain result.
+          (insert (if face (propertize message 'face face) message)))
         (princ "\n"))))
   (princ "\n"))
 
+(defun flycheck-get-next-checkers (checker)
+  "Return the immediate next checkers of CHECKER.
+
+This is a list of checker symbols.  The error levels of the
+`:next-checker' property are ignored."
+  (mapcar (lambda (c) (if (consp c) (cdr c) c))
+          (flycheck-checker-get checker 'next-checkers)))
+
+(defun flycheck-all-next-checkers (checker)
+  "Return all checkers that may follow CHECKER.
+
+Return the transitive closure of the next-checker relation.  The
+return value is a list of checkers, not including CHECKER."
+  (let ((next (flycheck-get-next-checkers checker)))
+    (seq-uniq (append next (seq-mapcat #'flycheck-all-next-checkers next)))))
+
+(defun flycheck--verify-next-checkers (checker)
+  "Return a verification result for the next checkers of CHECKER."
+  (-when-let (next (flycheck-get-next-checkers checker))
+    (list
+     (flycheck-verification-result-new
+      :label "next checkers"
+      ;; We use `insert-text-button' to preserve the button properties in
+      ;; the string
+      :message (string-join
+                (seq-map (lambda (checker)
+                           (make-text-button (symbol-name checker) nil
+                                             'type 'help-flycheck-checker-doc
+                                             'help-args (list checker)))
+                         next)
+                ", ")))))
```

Comment:
Not that I look at this again, I think this is an ideal candidate for mapconcat :)